### PR TITLE
DS-2578: Restore menu item `onClickHandler` support and add regression coverage

### PR DESCRIPTION
**Pull request by smorrisods**
## Summary
This PR restores `onClickHandler` support for header menu items and adds regression coverage for the affected behaviour.

## Changes
- fix click-handler wiring for header and overflow menu items
- add unit regression tests for subheader and overflow menu item `onClickHandler` execution
- add Playwright E2E tests for overflow menu browser behaviour:
  - opens on `menuButtonToggled: true`
  - closes on `menuButtonToggled: false`
  - closes when focus leaves in standalone mode
- remove duplicate overflow-menu unit tests now covered by E2E
- update the demo script in `src/index.html` to exercise menu item click handlers
- include required generated type updates:
  - `packages/ontario-design-system-component-library/src/components.d.ts`
  - `packages/ontario-design-system-component-library-react/src/types/stencil-hydrate.d.ts`
- include related build config fix:
  - `packages/ontario-design-system-component-library/stencil.config.ts` (`clientModule` path)

## Why
- restore expected callback behaviour for consumers using menu item click handlers
- prevent regression with targeted unit and E2E coverage
- keep generated/build artifacts aligned with source changes

## Validation
- unit tests updated for header click-handler paths
- Playwright E2E tests added for overflow menu interaction behaviour
- manual verification via updated demo script

### CODE REVIEW

File: packages/ontario-design-system-component-library/src/components/ontario-header-overflow-menu/test/ontario-header-overflow-menu.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+test.describe('ontario-header-overflow-menu', () => {
+	let host: Locator;
+	let nav: Locator;
+
+	test.beforeEach(async ({ page }) => {
+		await page.setContent('<ontario-header-overflow-menu></ontario-header-overflow-menu>');
+		await page.waitForChanges();
+
+		host = page.locator('ontario-header-overflow-menu').first();
+		nav = host.locator('nav#ontario-application-navigation');
```
Comment (erincarrieresontario) [inline packages/ontario-design-system-component-library/src/components/ontario-header-overflow-menu/test/ontario-header-overflow-menu.e2e.ts]: Since IDs are already unique, we don't need to specify the tag here

File: packages/ontario-design-system-component-library/src/components/ontario-header-overflow-menu/test/ontario-header-overflow-menu.e2e.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+				{ title: 'Link 1', href: '/link-1', linkIsActive: false },
+				{ title: 'Link 2', href: '/link-2', linkIsActive: false },
+			];
+		});
+		await page.waitForChanges();
+	});
+
+	test('opens menu when menuButtonToggled detail is true', async ({ page }) => {
+		await page.evaluate(() => {
+			window.dispatchEvent(new CustomEvent('menuButtonToggled', { detail: true }));
```
[the remark this answers is not in the document]
Comment (smorrisods) [reply packages/ontario-design-system-component-library/src/components/ontario-header-overflow-menu/test/ontario-header-overflow-menu.e2e.ts]: I'm actually going to look into this more tomorrow, apparently it is possible! 🕵️‍♂️

**Review by MeaghanCarrieres — APPROVED** [review]
